hpke.go

```diff
@@ -136,7 +136,14 @@ type hpkeContext struct {
 	infoHash  []byte `tls:"head=none"`
 }
 
-func keySchedule(suite CipherSuite, mode HPKEMode, pkR KEMPublicKey, zz, enc, info, psk, pskID, pkIm []byte) (key, nonce []byte, err error) {
+type contextParameters struct {
+	context hpkeContext
```

Comment:
I think you need at least one more field here, for the marshaled `context` value.  Maybe also the `keyContext` and `nonceContext` values.